### ISSUE TITLE
Command injection warning expected but not found

### ISSUE THREAD

**Issue by JacobEvelyn** [presidentbeef/brakeman#1179]
### Background

Brakeman version: 4.2.1
Rails version: 5.1
Ruby version: 2.5.1

### Issue

I noticed that `brakeman` correctly warns me about the shell injection vulnerability in something like the following:

```ruby
arguments = [method_that_returns_user_input, "--some-flag"].join(" ")

system("ruby #{arguments}")
```

However, I expect to see the same false-positive warning when I rearrange this code like so:

```ruby
command = ["ruby", method_that_returns_user_input, "--some-flag"].join(" ")

system(command)
```

and yet I do not. Is there any way to modify `brakeman`'s code to add this check? I'd be happy to help write a PR but I'd need some guidance on where to start.

**Comment by presidentbeef** [comment]
Hi Jacob,

Thank you for bringing this up! I can think of two ways to do this.

The simpler way would be to modify the current check to simply look for `Array#join` calls as arguments and then check the members of the array.

The other (possibly cooler) way would be to convert all `Array#join` calls to strings/string interpolation. 
For example:

```ruby
["ruby", method_that_returns_user_input, "--some-flag"].join(" ")
```

Would become

```ruby
"ruby #{method_that_returns_user_input} --some-flag"
```

I kind of like the second option better. It may help a lot of different checks to have this behavior.

**Comment by presidentbeef** [comment]
Seems like I forgot to mention this was addressed in #1198.
